src/services/edusharing/logic/connector.js

```diff
@@ -126,36 +121,62 @@ class EduSharingConnector {
 
 
 	async login() {
+		logger.info('Renewal of Edusharing credentials');
 		this.authorization = await this.getCookie();
 		this.accessToken = await this.getAuth();
 	}
 
-	isLoggedin() {
-		// returns false if cookie or accesstoken is falsy
-		return !!this.authorization && !!this.accessToken;
+	shouldRelogin() {
+		const nextCookieRenewalTime = lastCookieRenewalTime
+			? new Date(lastCookieRenewalTime.getTime() + COOKIE_RENEWAL_PERIOD_MS)
+			: new Date();
+		// should relogin if cookie expired or cookie or access token is empty
+		const shouldRelogin = (new Date() >= nextCookieRenewalTime) || !this.authorization || !this.accessToken;
+		if (shouldRelogin) {
+			lastCookieRenewalTime = new Date();
+		}
+		return shouldRelogin;
 	}
 
-	async getImage(url) {
+	async getImage(url, retry = true) {
 		const reqOptions = {
 			uri: url,
 			method: 'GET',
 			headers: {},
 			// necessary to get the image as binary value
 			encoding: null,
+			resolveWithFullResponse: true,
 		};
 		return request(reqOptions)
-			.then((result) => {
-				const encodedData = `data:image;base64,${result.toString('base64')}`;
+			.then(async (result) => {
+				// edusharing sometimes doesn't return error code if access token is expired
+				// but redirect to no permission image
+				if (retry && result.req.path === NO_PERMISSIONS_IMG) {
+					return this.tryToGetImageWithNewAccessToken(url);
+				}
+				const encodedData = `data:image;base64,${result.body.toString('base64')}`;
 				return Promise.resolve(encodedData);
 			})
-			.catch((err) => {
-				logger.error('error: ', err);
+			.catch(async (err) => {
+				if (retry && RETRY_ERROR_CODES.indexOf(err.statusCode) >= 0) {
+					return this.tryToGetImageWithNewAccessToken(url);
+				}
+				logger.error(`Couldn't fetch image for ${url}:
+				Edusharing responded with ${err.statusCode} ${err.message}`);
 				return Promise.reject(err);
 			});
 	}
 
+	async tryToGetImageWithNewAccessToken(url) {
+		logger.info('Trying to renew Edusharing access token.');
+		this.accessToken = await this.getAuth();
```

Comment:
should we call shouldRelogin and login methods here, instead of forcing a refresh of the token alone?
is there a link between cookie and tocken?

Reply:
cookie and login are not related to this place. The image cannot be loaded because accessToken expires. 
We have additional mechanism to refresh cookies and accessToken once per 30 minutes.